packages/ui/client/components/views/ViewReport.cy.tsx

```diff
@@ -0,0 +1,120 @@
+import ViewReport from './ViewReport.vue'
+import type { File } from '#types'
+
+const taskErrorSelector = '.task-error'
+
+describe('ViewReport', () => {
+  it('test plain stack trace', () => {
+    const file: File = {
+      id: 'f-1',
+      name: 'test/plain-stack-trace.ts',
+      type: 'suite',
+      mode: 'run',
+      filepath: 'test/plain-stack-trace.ts',
+      result: {
+        state: 'fail',
+        error: {
+          name: 'Do some test',
+          stacks: [{ line: 10, column: 20, file: 'test/plain-stack-trace.ts', method: 'dummy test' }],
+          message: 'Error: Transform failed with 1 error:',
+        },
+      },
+      tasks: [],
+    }
+    const container = cy.mount(<ViewReport file={file} />)
```

Comment:
Do we need TSX here? Possible to simply use `h`?

Reply:
it is the cypress test, I just "clone" the existing one

Reply:
you have it on dashboard folder (the original from Jess)

Reply:
@antfu I can change cy tests to render the component and passing the props instead using tsx

Reply:
```ts
const container = cy.mount(ViewReport, { 
  propsData: { file } 
})
```

Reply:
> Do we need TSX here? Possible to simply use `h`?

The reason we use TSX in such a simple example is for consistency. I prefer native `h`, but I think TSX is the most elegant answer right now to:
1. passing in spies for emitted events
2. making components under test reactive/into playgrounds
3. styling wrapper or related content

The other pattern we explored at Cy was using SFCs.

### Passing Spies

```tsx
const onMyEventSpy = cy.spy().as('myEventSpy')
mount(() => <MyComponent onMyEvent={onMyEventSpy}>)

cy.get('button').click().get('@myEventSpy').should('have.been.called.with', 'some arguments')
```

### Interactive Tests

TSX becomes really helpful for modals when you not only want to **spy** on the event emitted (somewhat identical to `wrapper.emitted()`, but also when you want to make the component into more of a playground and **do** something. e.g.

```tsx
const visible = ref(true)
mount(() => (
  <button onClick={visible.value = !visible.value}>Toggle</button>
  <MyModal visible={visible}/>
))
```

### Styling Wrapper DOM

Lastly, when you want to control parent/container styles in order to see your component behave differently in different UIs. This works especially well with UnoCSS

```tsx
mount(() => (
  <div min-w-200px resize overflow-auto>
    <MyTextComponent>Content inside here!</MyTextComponent>
  </div>
))
```